docs/09_tutorials/10_nested_containers.md

```diff
@@ -0,0 +1,199 @@
+---
+content_title: Nested Containers
+---
+
+## Overview
+
+This tutorial demonstrates the usage of various C++ nested-containers in smart contracts for the following EOSIO types:
+
+* [Multi-Index Table](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/multi-index/index)
+* [Key-Value Map](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/key-value-api/kv_map/index)
+* [Key-Value Table](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/key-value-api/kv_table/index) 
+
+### The Nested-Containers Covered Are
+
+This tutorial covers the following netsted-containers combinations.
+
+* Single standard template library (STL) containers defined in the `std` namespace
+    * std::vector
+    * std::set
+    * std::optional
+    * std::map
+    * std::tuple
+    * std::pair which needs special attention for cleos input format
+* Standard C++ structures
+    * Simple struct
+    * Complex struct
+* Two-layers nested containers made of combinations of the following
+    * std::vector
+    * std::set
+    * std::optional
+    * std::map
+    * std::tuple
+    * std::pair
+* One three-layers nested container for farther extentions demonstration
+
+### Cleos commands
+
+In adition the cleos commands for various nested-containers are also documented. Specifically it shows how to emplace and how to retrieve data using cleos *get table* and *get kv_table*. The cleos commands are documented at the end of the two shell scripts which accompany this tutorial.
+
```

Comment:
I need to discuss with Qing to find good locations of shell scripts, and also the locations of *.cpp files next week, please remind me to do this, so that you have a clear picture about the URLs you can plug in for this documentation:)